src/assets/SingleArticle.jsx

```diff
@@ -1,33 +1,66 @@
 import React, { useEffect, useState } from "react";
 import axios from "axios";
 import { useParams } from "react-router-dom";
-import "./CSS/SingleArticle.css"
+import "./CSS/SingleArticle.css";
 import Comments from "./Comments";
 
 export default function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState(false);
+  const [votes, setVotes] = useState(0);
+  const [voted, setVoted] = useState(false);
 
   useEffect(() => {
     setIsLoading(true);
     setIsError(false);
     axios
       .get(`https://js-be-project.onrender.com/api/articles/${article_id}`)
       .then((apiResponse) => {
-        setArticle(apiResponse.data.articles[0]);
+        const articleData = apiResponse.data.articles[0];
+        setArticle(articleData);
+        setVotes(articleData.votes);
         setIsLoading(false);
       })
       .catch(() => {
         setIsError(true);
         setIsLoading(false);
       });
   }, [article_id]);
+
+  const handleVote = () => {
+    if (!voted) {
+      setVotes((currentVotes) => currentVotes + 1);
+      setVoted(true);
+      axios
+        .patch(
+          `https://js-be-project.onrender.com/api/articles/${article_id}`,
+          { inc_votes: 1 }
+        )
+        .catch((err) => {
+          setVotes((currentVotes) => currentVotes - 1);
+          console.log(err);
+          setVoted(false);
+        });
+    } else {
+      setVotes((currentVotes) => currentVotes - 1);
+      setVoted(false);
+      axios
+        .patch(
+          `https://js-be-project.onrender.com/api/articles/${article_id}`,
+          { inc_votes: -1 }
+        )
```

Comment:
You could have a look at extracting out these calls into another file and just calling the function here, might clean this section up a bit